addons/mail/static/tests/helpers/test_utils.js

```diff
@@ -594,16 +594,18 @@ export const click = getClick({ afterNextRender });
  * @param {string} selector
  */
 export function waitUntil(selector) {
-    return new Promise((resolve) => {
+    return new Promise((resolve, reject) => {
         const $selector = $(selector);
-        if ($(selector)) {
+        if ($selector.length) {
             return resolve($selector);
         }
 
+        const timer = setTimeout(reject(new Error("Waited 5 second for " + selector)), 5000);
         const observer = new MutationObserver((mutations) => {
-            if ($selector) {
+            if ($selector.length) {
```

Comment:
Does that recompute the selector?
Otherwise the `$` method should be called again instead of storing it.